compiler/daml-lf-tools/src/DA/Daml/LF/Simplifier.hs

```diff
@@ -353,14 +387,76 @@ simplifyExpr world = fst . cata go
       -- - If `let x = e1 in e2` is k-safe, then `e1` is 0-safe and `e2` is
       --   k-safe.
       EStructProjF f (ELet (Binding (x, t) e1) e2, Info fv sf _) ->
-        go $ ELetF (BindingF (x, t) (e1, s1)) (go $ EStructProjF f (e2, s2))
+        go world $ ELetF (BindingF (x, t) (e1, s1)) (go world $ EStructProjF f (e2, s2))
         where
           s1 = Info fv (sf `min` Safe 0) TCNeither
           s2 = Info (freeExprVar x <> fv) sf TCNeither
 
       -- e    ==>    e
       e -> (embed (fmap fst e), infoStep world (fmap snd e))
 
+isWorthLifting :: Expr -> Bool
```

Comment:
A comment here that explains why we don’t lift everything would be nice. I.e., what are the performance tradeoffs, do we loose performance by lifting things that are “too small”, if so have we benchmarked that this is the right tradeoff, …?